integration/auth/user_mgt_test.go

```diff
@@ -91,6 +92,219 @@ func TestGetNonExistingUser(t *testing.T) {
+		ProviderData([](*auth.UserProvider){
+			&auth.UserProvider{
+				ProviderID: "google.com",
+				UID:        "google_" + importUser1UID,
+			},
+		})
+
+	createTestUsers := func(t *testing.T) {
+		testUser1 = newUserWithParams(t)
+		testUser2 = newUserWithParams(t)
+		testUser3 = newUserWithParams(t)
+		importUser(t, importUser1UID, importUser1)
+	}
+
+	// Checks to see if the users list contain the given uids. Order is ignored.
+	//
+	// Behaviour is undefined if there are duplicate entries in either of the
+	// slices.
+	//
+	// This function is identical to the one in auth/user_mgt_test.go
+	sameUsers := func(users [](*auth.UserRecord), uids []string) bool {
+		if len(users) != len(uids) {
+			return false
+		}
+
+		sort.Slice(users, func(i, j int) bool {
+			return users[i].UID < users[j].UID
+		})
+		sort.Slice(uids, func(i, j int) bool {
+			return uids[i] < uids[j]
+		})
+
+		for i := range users {
+			if users[i].UID != uids[i] {
+				return false
+			}
+		}
+
+		return true
+	}
+
+	createTestUsers(t)
+	defer deleteUser(testUser1.UID)
+	defer deleteUser(testUser2.UID)
+	defer deleteUser(testUser3.UID)
+	defer deleteUser(importUser1UID)
+
+	userRecordsToUIDs := func(users [](*auth.UserRecord)) []string {
+		results := []string{}
+		for i := range users {
+			results = append(results, users[i].UID)
+		}
+		return results
+	}
+
+	t.Run("returns users by various identifier types in a single call", func(t *testing.T) {
+		getUsersResult, err := client.GetUsers(context.Background(), []auth.UserIdentifier{
+			auth.UIDIdentifier{UID: testUser1.UID},
+			auth.EmailIdentifier{Email: testUser2.Email},
+			auth.PhoneIdentifier{PhoneNumber: testUser3.PhoneNumber},
+			auth.ProviderIdentifier{ProviderID: "google.com", ProviderUID: "google_" + importUser1UID},
+		})
+		if err != nil {
+			t.Fatalf("GetUsers([valid identifiers]) returned an error: %v", err)
+		}
+
+		if !sameUsers(getUsersResult.Users, []string{testUser1.UID, testUser2.UID, testUser3.UID, importUser1UID}) {
+			t.Errorf("GetUsers([valid identifiers]) = %v; want = %v (in any order)",
+				userRecordsToUIDs(getUsersResult.Users), []string{testUser1.UID, testUser2.UID, testUser3.UID, importUser1UID})
+		}
+	})
+
+	t.Run("returns found users and ignores non-existing users", func(t *testing.T) {
+		getUsersResult, err := client.GetUsers(context.Background(), []auth.UserIdentifier{
+			auth.UIDIdentifier{UID: testUser1.UID},
+			auth.UIDIdentifier{UID: "uid_that_doesnt_exist"},
+			auth.UIDIdentifier{UID: testUser3.UID},
+		})
+		if err != nil {
+			t.Fatalf("GetUsers([...]) returned an error: %v", err)
+		}
+
+		if !sameUsers(getUsersResult.Users, []string{testUser1.UID, testUser3.UID}) {
+			t.Errorf("GetUsers([valid identifiers]) = %v; want = %v (in any order)",
+				getUsersResult.Users, []string{testUser1.UID, testUser3.UID})
+		}
+		if len(getUsersResult.NotFound) != 1 {
+			t.Errorf("len(GetUsers([...]).NotFound) = %d; want 1", len(getUsersResult.NotFound))
```

Comment:
t.Fatalf?

Reply:
This was deliberate (though also somewhat pedantic). I can clearly use Fatalf now to eliminate the else, but if the GetUsersResult struct gets an extra field added to it and a corresponding block added to this test, then the Fatalf would needlessly stop the test before that extra field could be verified.